google/cloud/bigquery/types/standard_sql.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+import enum
+from typing import Any, Dict, Iterable, Optional
+
+
+@dataclass
+class StandardSqlDataType:
+    """The type of a variable, e.g., a function argument.
+
+    See:
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/StandardSqlDataType
+
+    Examples:
+
+    .. code-block:: text
+
+        INT64: {type_kind="INT64"}
+        ARRAY: {type_kind="ARRAY", array_element_type="STRING"}
+        STRUCT<x STRING, y ARRAY>: {
+            type_kind="STRUCT",
+            struct_type={
+                fields=[
+                    {name="x", type={type_kind="STRING"}},
+                    {
+                        name="y",
+                        type={type_kind="ARRAY", array_element_type="DATE"}
+                    }
+                ]
+            }
+        }
+    """
+
+    class TypeKind(str, enum.Enum):
```

Comment:
We could use plain strings, but they are susceptible to typos, thus I'd rather use `TypeKind.INT64` instead of `"INT64"` in the code, not to mention cases where a (non-existing) `"INTEGER"` string literal is used (actually happened to me a few times).

Presumably the set of available SQL types does not really change often?

Reply:
Another thought - if we decide to keep the `TypeKind` enum, I would unnest it. Being defined under the `StandardSqlDataType`, it makes the code that uses it unnecessarily verbose.